### PR TITLE
Add Django 5.0 support

### PR DESCRIPTION
**Pull request by pfouque**
Add Django 5.0 support to the test matrix and package description

Type: Feature

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+    py{38,39,310,311}-django41-psycopg2
+    py{38,39,310}-django32-psycopg2
+    report
+
 
 [testenv]
 install_command = pip install {opts} --no-compile {packages}
 deps =
-    django32: Django>=3.2,<3.3
+    django50: Django==5.0rc1
```
Comment (adamchainz) [inline tox.ini:16]:
```suggestion
    django50: Django>=5.0rc1,<5.1
```

to match the others

File: tox.ini [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
     django42: Django>=4.2,<4.3
+    django41: Django>=4.1,<4.2
+    django32: Django>=3.2,<3.3
```
Comment (adamchainz) [inline tox.ini:19]: Weird to leave Django versions in descending order but pscyopg versions in ascending order. I think sticking to ascending order is the least invasive, here and in `envlist`. (`tox-ini-fmt` sorts descending, if that would ever be run on this project.)

**Comment by adamchainz** [comment]
Also just spotted that on Django 5.0, `ProxyField` can be replaced with [the new `GeneratedField`](https://www.paulox.net/2023/11/07/database-generated-columns-part-1-django-and-sqlite/). Perhaps one to deprecate later, since it is tracked in migrations, unlike `ProxyField`.

**Comment by pfouque** [comment]
Thanks!

**Pull request closed without merging**